shopify_python/google_styleguide.py

```diff
@@ -251,3 +256,45 @@ def __use_cond_expr(self, node):  # type: (astroid.If) -> None
                 else_body_name = node.orelse[0].targets[0].name
                 if if_body_name == else_body_name:
                     self.add_message('cond-expr', node=node)
+
+    def __lambda_func(self, node):  # type: (astroid.Lambda) -> None
+        """Prefer Operator Function to Lambda Functions"""
+
+        if isinstance(node.body, astroid.UnaryOp):
+            unary_operators = {
+                "~": "invert",
+                "-": "neg",
+                "not": "not_",
+                "+": "pos"
+            }
+            operator = unary_operators.get(node.body.op)
+            argname = node.args.args[0].name
+            if operator and not isinstance(node.body.operand, astroid.BinOp) and argname is node.body.operand.name:
```

Comment:
The check for a `BinOp` operand seems fragile to me. Is that the only possible operand, besides a variable reference? Couldn't it be another unary operation? Or a function call?

IIUC the node type you actually expect is `astroid.Name`. Maybe look for that explicitly.